### PR TITLE
Add option to print output of submit in json.

### LINKED ISSUE #604: Support argo submit output to be in json format
Argo submit shows the result of the command in a nice tabular format. However, for running `argo submit` in a script, it is a bit hard to parse the output from the tabular format. Instead, it'll be good to have the output of `argo submit` in the json format.

### PR DESCRIPTION
**Pull request by shrinandj**
```
$ ./dist/argo submit examples/hello-world.yaml --output json
{
    "metadata": {
        "name": "hello-world-jgnhz",
        "generateName": "hello-world-",
        "namespace": "default",
        "selfLink": "/apis/argoproj.io/v1alpha1/namespaces/default/workflows/hello-world-jgnhz",
        "uid": "aaf4e8f2-e1f8-11e7-8dd9-42010a8a00c4",
        "resourceVersion": "320353",
        "creationTimestamp": "2017-12-16T00:32:59Z"
    },
    "spec": {
        "templates": [
            {
                "name": "whalesay",
                "inputs": {},
                "outputs": {},
                "container": {
                    "name": "",
                    "image": "docker/whalesay:latest",
                    "command": [
                        "cowsay"
                    ],
                    "args": [
                        "hello world"
                    ],
                    "resources": {}
                }
            }
        ],
        "entrypoint": "whalesay",
        "arguments": {}
    },
    "status": {
        "phase": "",
        "startedAt": null,
        "finishedAt": null,
        "nodes": null
    }
}
$ ./dist/argo submit examples/hello-world.yaml
Name:             hello-world-rrn5s
Namespace:        default
Status:           Pending
Created:          Fri Dec 15 16:33:02 -0800 (now)
$ ./dist/argo submit examples/hello-world.yaml --output json | jq .metadata.name
"hello-world-rsljt"
```

Closes #604.

### CODE REVIEW

**Comment by jessesuen** [comment]
What happens when you `argo submit` multiple workflows and pipe to `jq`?

**Comment by shrinandj** [comment]
Works just fine ...

```
 ./dist/argo submit examples/hello-world.yaml examples/conditionals.yaml examples/coinflip.yaml --output json | jq .metadata.name
"hello-world-z7swh"
"conditional-mnb7h"
"coinflip-w25gv"
```

**Comment by jessesuen** [comment]
Cool!

**Comment by jessesuen** [comment]
I just learned that kubectl has an `kubectl create wf.yaml -o name` option that I think we should replicate. See example usage:
```
$ kubectl create -f examples/hello-world.yaml -o name
workflow/hello-world-4xkpb
```

This would help a lot of scripting, especially if `jq` is unavailable. Even if `jq` is present to consume the `-o json`, there are still the double quotes to deal with.

**Comment by shrinandj** [comment]
You mean, there should be `argo submit -o json` as well as `argo submit -o name` that will output detailed json and simply the name respectively, right?

**Comment by jessesuen** [comment]
Right.

**Comment by shrinandj** [comment]
Add the "-o name" option.

**Comment by jessesuen** [comment]
LGTM
